go-controller/pkg/node/node.go

```diff
@@ -595,6 +604,75 @@ func (n *OvnNode) WatchEndpoints() error {
 	return err
 }
 
+func exGatewayPodsAnnotationsChanged(oldNs, newNs *kapi.Namespace) bool {
+	// We only care about exgw pod deletions, i.e if the IPs on a namespace reduced than before, we return true
+	// NOTE: we don't care about the whole namespace getting deleted since pods will go away and so will the conntrack entries
+	return len(oldNs.Annotations[util.ExternalGatewayPodIPsAnnotation]) > len(newNs.Annotations[util.ExternalGatewayPodIPsAnnotation])
```

Comment:
@tssurya this seems a bit fragile; it won't happen now due to the way we add/remove single IPs at a time, but I don't think we should rely on that. Should we instead do a strings.Split() and then compare lengths of the resulting arrays instead?

Reply:
yup excellent point, what if its dual stack and IPV6 addresses lengths could mess up depending on how they look.. let me change this with what you suggested. thanks Dan as usual for your hawk eyes :)

Reply:
@tssurya ideally we compare the actual contents too... but maybe that's not necessary right now given our single-ip-change behavior. But maybe we should just do the full compare now and be done with it... there won't be that many IPs so I don't think it would be very heavy.

Reply:
hmm ok so thinking more about this, first step is to get the list of IPs from new annotation -> with the split done - and check if the "number of IPs" is less than before. but then you are right in the sense what if an IP got added and deleted at same time like with deployments where another one takes the place somehow and the length of my annotation stays the same....

I could do what you are suggesting, like really do a full blown comparision, would be as simple as old != new ? then trigger the arping etc.. but problem is we will do this for every SPK add as well, not just delete. Is that acceptable?

Reply:
ACK let me change this logic in code per Dan's suggestions ^